.github/workflows/python-unit-test.yml

```diff
@@ -20,8 +20,33 @@ on:
     types: [opened, synchronize, reopened]
 
 jobs:
+  prepare-deps:
+    runs-on: ubuntu-latest
+    env:
+      RAPIDS_USER_TOOLS_CACHE_FOLDER: ${{ github.workspace }}/.cache/spark_rapids_user_tools_cache
+    steps:
+      - name: Checkout code
+        uses: NVIDIA/spark-rapids-common/checkout@main
+
+      - name: Prefetch heavy on-prem dependency artifacts
+        shell: bash
+        run: |
+          chmod +x scripts/ci_prefetch_onprem_deps.sh
+          scripts/ci_prefetch_onprem_deps.sh "$RAPIDS_USER_TOOLS_CACHE_FOLDER" "user_tools/src/spark_rapids_pytools/resources/onprem-configs.json"
```

Comment:
I understand that we have most of our python-tests defaulting to on-prem. Just for sake of future expansions, what would be the best way to support different platforms or multiple platforms?